cmake fails  on Ubuntu 20.04

Some cmake guidance would be very helpful. Here are the results of my attempt:

 cmake -DBUILD_SHARED_LIBS=OFF -DENABLE_EXAMPLES=ON -G "Unix Makefiles"
 .
-- The C compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for arpa/inet.h
-- Looking for arpa/inet.h - found
-- Looking for dlfcn.h
-- Looking for dlfcn.h - found
-- Looking for fuse.h
-- Looking for fuse.h - not found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for memory.h
-- Looking for memory.h - found
-- Looking for netdb.h
-- Looking for netdb.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for netinet/tcp.h
-- Looking for netinet/tcp.h - found
-- Looking for net/if.h
-- Looking for net/if.h - found
-- Looking for poll.h
-- Looking for poll.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for strings.h
-- Looking for strings.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for sys/filio.h
-- Looking for sys/filio.h - not found
-- Looking for sys/ioctl.h
-- Looking for sys/ioctl.h - found
-- Looking for sys/socket.h
-- Looking for sys/socket.h - found
-- Looking for sys/statvfs.h
-- Looking for sys/statvfs.h - found
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for sys/sysmacros.h
-- Looking for sys/sysmacros.h - found
-- Looking for sys/time.h
-- Looking for sys/time.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for sys/vfs.h
-- Looking for sys/vfs.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for utime.h
-- Looking for utime.h - found
-- Performing Test HAVE_SOCKADDR_LEN
-- Performing Test HAVE_SOCKADDR_LEN - Failed
-- Performing Test HAVE_SOCKADDR_STORAGE
-- Performing Test HAVE_SOCKADDR_STORAGE - Success
-- Performing Test HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC
-- Performing Test HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC - Success
-- Performing Test HAVE_SO_BINDTODEVICE
-- Performing Test HAVE_SO_BINDTODEVICE - Success
-- Performing Test HAVE_TALLOC_TEVENT
-- Performing Test HAVE_TALLOC_TEVENT - Failed
-- Performing Test HAVE_CLOCK_GETTIME
-- Performing Test HAVE_CLOCK_GETTIME - Success
-- Performing Test NO_LFS_REQUIRED
-- Performing Test NO_LFS_REQUIRED - Success
-- Looking for makedev
-- Looking for makedev - not found
-- Looking for makedev
-- Looking for makedev - found
-- Performing Test C_ACCEPTS_WALL
-- Performing Test C_ACCEPTS_WALL - Success
-- Configuring done
CMake Error: install(EXPORT "libnfs" ...) includes target "nfs" which requires target "nfs_rquota" that is not in any export set.
CMake Error: install(EXPORT "libnfs" ...) includes target "nfs" which requires target "nfs_portmap" that is not in any export set.
CMake Error: install(EXPORT "libnfs" ...) includes target "nfs" which requires target "nfs_nsm" that is not in any export set.
CMake Error: install(EXPORT "libnfs" ...) includes target "nfs" which requires target "nfs_nlm" that is not in any export set.
CMake Error: install(EXPORT "libnfs" ...) includes target "nfs" which requires target "nfs_nfs4" that is not in any export set.
CMake Error: install(EXPORT "libnfs" ...) includes target "nfs" which requires target "nfs_nfs" that is not in any export set.
CMake Error: install(EXPORT "libnfs" ...) includes target "nfs" which requires target "nfs_mount" that is not in any export set.
CMake Error in CMakeLists.txt:
  export called with target "nfs" which requires target "nfs_rquota" that is
  not in any export set.


CMake Error in CMakeLists.txt:
  export called with target "nfs" which requires target "nfs_portmap" that is
  not in any export set.


CMake Error in CMakeLists.txt:
  export called with target "nfs" which requires target "nfs_nsm" that is not
  in any export set.


CMake Error in CMakeLists.txt:
  export called with target "nfs" which requires target "nfs_nlm" that is not
  in any export set.


CMake Error in CMakeLists.txt:
  export called with target "nfs" which requires target "nfs_nfs4" that is
  not in any export set.


CMake Error in CMakeLists.txt:
  export called with target "nfs" which requires target "nfs_nfs" that is not
  in any export set.


CMake Error in CMakeLists.txt:
  export called with target "nfs" which requires target "nfs_mount" that is
  not in any export set.


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.

Just confirming this failure as well, macos host/target. 5.0.0 tarball release

```
cd macosx12.0_x86_64-target-debug/build; /Users/Shared/xbmc-depends/arm-darwin21.2.0-native/bin/cmake -DCMAKE_TOOLCHAIN_FILE=/Users/Shared/xbmc-depends/macosx12.0_x86_64-target-debug/share/Toolchain.cmake -DCMAKE_INSTALL_PREFIX=/Users/Shared/xbmc-depends/macosx12.0_x86_64-target-debug -DBUILD_SHARED_LIBS=OFF -DENABLE_TESTS=OFF -DENABLE_DOCUMENTATION=OFF -DENABLE_UTILS=OFF -DENABLE_EXAMPLES=OFF ..
-- The C compiler identification is AppleClang 13.0.0.13000029
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for arpa/inet.h
-- Looking for arpa/inet.h - found
-- Looking for dlfcn.h
-- Looking for dlfcn.h - found
-- Looking for fuse.h
-- Looking for fuse.h - not found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for memory.h
-- Looking for memory.h - found
-- Looking for netdb.h
-- Looking for netdb.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for netinet/tcp.h
-- Looking for netinet/tcp.h - found
-- Looking for net/if.h
-- Looking for net/if.h - found
-- Looking for poll.h
-- Looking for poll.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for strings.h
-- Looking for strings.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for sys/filio.h
-- Looking for sys/filio.h - found
-- Looking for sys/ioctl.h
-- Looking for sys/ioctl.h - found
-- Looking for sys/socket.h
-- Looking for sys/socket.h - found
-- Looking for sys/statvfs.h
-- Looking for sys/statvfs.h - found
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for sys/sysmacros.h
-- Looking for sys/sysmacros.h - not found
-- Looking for sys/time.h
-- Looking for sys/time.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for sys/vfs.h
-- Looking for sys/vfs.h - not found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for utime.h
-- Looking for utime.h - found
-- Performing Test HAVE_SOCKADDR_LEN
-- Performing Test HAVE_SOCKADDR_LEN - Success
-- Performing Test HAVE_SOCKADDR_STORAGE
-- Performing Test HAVE_SOCKADDR_STORAGE - Success
-- Performing Test HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC
-- Performing Test HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC - Failed
-- Performing Test HAVE_SO_BINDTODEVICE
-- Performing Test HAVE_SO_BINDTODEVICE - Failed
-- Performing Test HAVE_TALLOC_TEVENT
-- Performing Test HAVE_TALLOC_TEVENT - Failed
-- Performing Test HAVE_CLOCK_GETTIME
-- Performing Test HAVE_CLOCK_GETTIME - Failed
-- Performing Test NO_LFS_REQUIRED
-- Performing Test NO_LFS_REQUIRED - Success
-- Looking for makedev
-- Looking for makedev - not found
-- Looking for makedev
-- Looking for makedev - not found
-- Configuring done
CMake Error: install(EXPORT "libnfs" ...) includes target "nfs" which requires target "nfs_rquota" that is not in any export set.
CMake Error: install(EXPORT "libnfs" ...) includes target "nfs" which requires target "nfs_portmap" that is not in any export set.
CMake Error: install(EXPORT "libnfs" ...) includes target "nfs" which requires target "nfs_nsm" that is not in any export set.
CMake Error: install(EXPORT "libnfs" ...) includes target "nfs" which requires target "nfs_nlm" that is not in any export set.
CMake Error: install(EXPORT "libnfs" ...) includes target "nfs" which requires target "nfs_nfs4" that is not in any export set.
CMake Error: install(EXPORT "libnfs" ...) includes target "nfs" which requires target "nfs_nfs" that is not in any export set.
CMake Error: install(EXPORT "libnfs" ...) includes target "nfs" which requires target "nfs_mount" that is not in any export set.
CMake Error in CMakeLists.txt:
  export called with target "nfs" which requires target "nfs_rquota" that is
  not in any export set.


CMake Error in CMakeLists.txt:
  export called with target "nfs" which requires target "nfs_portmap" that is
  not in any export set.


CMake Error in CMakeLists.txt:
  export called with target "nfs" which requires target "nfs_nsm" that is not
  in any export set.


CMake Error in CMakeLists.txt:
  export called with target "nfs" which requires target "nfs_nlm" that is not
  in any export set.


CMake Error in CMakeLists.txt:
  export called with target "nfs" which requires target "nfs_nfs4" that is
  not in any export set.


CMake Error in CMakeLists.txt:
  export called with target "nfs" which requires target "nfs_nfs" that is not
  in any export set.


CMake Error in CMakeLists.txt:
  export called with target "nfs" which requires target "nfs_mount" that is
  not in any export set.


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
make: *** [macosx12.0_x86_64-target-debug] Error 1
```

--- a/cmake/Macros.cmake
+++ b/cmake/Macros.cmake
@@ -12,4 +12,12 @@ function(core_add_library name)
   add_library(${name} OBJECT ${SOURCES} ${HEADERS})
   target_include_directories(${name} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
   set(CORE_LIBRARIES "${name};${CORE_LIBRARIES}" CACHE INTERNAL "")
+
+  # no need to install core libs if we build shared library
+  if(NOT BUILD_SHARED_LIBS)
+    install(TARGETS ${name} EXPORT libnfs
+            RUNTIME DESTINATION bin
+            ARCHIVE DESTINATION lib
+            LIBRARY DESTINATION lib)
+  endif()
 endfunction()

